ra2ce/analysis/analysis_factory.py

```diff
@@ -51,20 +69,105 @@ def get_analysis(self, analysis_config: AnalysisConfigWrapper) -> AnalysisProtoc
+            )
+        if self.analysis.analysis == AnalysisIndirectEnum.MULTI_LINK_REDUNDANCY:
+            return MultiLinkRedundancy(
+                graph_file=analysis_config.graph_files.base_graph_hazard, **_input_dict
+            )
+        if (
+            self.analysis.analysis
+            == AnalysisIndirectEnum.OPTIMAL_ROUTE_ORIGIN_DESTINATION
+        ):
+            return OptimalRouteOriginDestination(
+                graph_file=analysis_config.graph_files.origins_destinations_graph,
+                **_input_dict,
+                origins_destinations=analysis_config.config_data.origins_destinations,
+            )
+        if self.analysis.analysis == AnalysisIndirectEnum.MULTI_LINK_ORIGIN_DESTINATION:
+            return MultiLinkOriginDestination(
+                graph_file=analysis_config.graph_files.origins_destinations_graph_hazard,
+                **_input_dict,
+                origins_destinations=analysis_config.config_data.origins_destinations,
+            )
+        if (
+            self.analysis.analysis
+            == AnalysisIndirectEnum.OPTIMAL_ROUTE_ORIGIN_CLOSEST_DESTINATION
+        ):
+            return OptimalRouteOriginClosestDestination(
+                graph_file=analysis_config.graph_files.origins_destinations_graph,
+                **_input_dict,
+                origins_destinations=analysis_config.config_data.origins_destinations,
+                file_id=analysis_config.config_data.network.file_id,
+            )
+        if (
+            self.analysis.analysis
+            == AnalysisIndirectEnum.MULTI_LINK_ORIGIN_CLOSEST_DESTINATION
+        ):
+            return MultiLinkOriginClosestDestination(
+                graph_file=analysis_config.graph_files.origins_destinations_graph,
+                graph_file_hazard=analysis_config.graph_files.origins_destinations_graph_hazard,
+                **_input_dict,
+                origins_destinations=analysis_config.config_data.origins_destinations,
+                file_id=analysis_config.config_data.network.file_id,
+            )
+        if self.analysis.analysis == AnalysisIndirectEnum.LOSSES:
+            return Losses(
+                graph_file=analysis_config.graph_files.base_graph_hazard, **_input_dict
+            )
+        if self.analysis.analysis == AnalysisIndirectEnum.SINGLE_LINK_LOSSES:
+            return SingleLinkLosses(
+                graph_file=analysis_config.graph_files.base_graph_hazard, **_input_dict
+            )
+        if self.analysis.analysis == AnalysisIndirectEnum.MULTI_LINK_LOSSES:
+            return MultiLinkLosses(
+                graph_file=analysis_config.graph_files.base_graph_hazard, **_input_dict
+            )
+        if self.analysis.analysis == AnalysisIndirectEnum.MULTI_LINK_ISOLATED_LOCATIONS:
+            return MultiLinkIsolatedLocations(
+                graph_file=analysis_config.graph_files.base_graph_hazard, **_input_dict
```

Comment:
I believe we can work together to make this more readable (preferably in a follow-up issue)

Reply:
Perfect

Reply:
Discussed as next steps:
* create subpackages for different types of analyses
* create private methods for the creation of the different analyses